pkg/cloud/aws/services/ec2/vpc_ip_resource_limit.go

```diff
@@ -0,0 +1,459 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ec2
+
+// copy from github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/awsutils/vpc_ip_resource_limit.go
+
+// InstanceENIsAvailable contains a mapping of instance types to the number of ENIs available which is described at
+// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html#AvailableIpPerENI
+var InstanceENIsAvailable = map[string]int{
+	"a1.medium":     2,
```

Comment:
Is it possible to generalize this a bit instead of a hardcoded map? For example it looks like `*.large` is 3 with the exception of `m4.large` which is 4.

Reply:
I think this is taken directly from https://github.com/aws/amazon-vpc-cni-k8s/blob/0c5e5b9454f2896001037e6468f9cfecaca647aa/pkg/awsutils/vpc_ip_resource_limit.go

I wonder if we could just import it?